src/app/guards/TenantGuard.tsx

```diff
@@ -21,29 +19,10 @@ function TenantGuard({ children }: BaseComponentProps) {
         [showTenantCreation]
     );
 
-    const {
-        userGrants,
-        isValidating: checkingGrants,
-        mutate,
-        error,
-    } = useUserGrants(true);
+    const hasAnyAccess = useUserInfoSummaryStore((state) => state.hasAnyAccess);
```

Comment:
`hasAccess` or, preferably, `accessExists` would be a better name for this state property IMO.